extensions/amp-consent/amp-consent.md

```diff
@@ -160,6 +160,10 @@ In the case that `checkConsentHref` and `promptIfUnknownForGeoGroup` are both de
 
 `promptUI`: Specifies the prompt element that is shown to collect the user's consent. The prompt element should be child element of `<amp-consent>` with an `id` that is referenced by the `promptUI`. See the [Prompt UI](#prompt-ui) section for details on how a user interacts with the prompt UI.
 
+#### stripPrefix
+
+`stripPrefix`: Causes `<amp-consent>` to strip a prefix from the `checkConsentHref` endpoint's return value. If the prefix is not present in the response, then this option will have no effect. `stripPrefix` can be useful for APIs that include [security prefixes](http://patorjk.com/blog/2013/02/05/crafty-tricks-for-avoiding-xssi/) like `)]}` to help prevent cross site scripting attacks.
```

Comment:
shall we always strip prefix?

Reply:
to clarify. we could always try to strip a set of common prefixes, such as `)]}` & `while(1)`. for endpoints that want to be xssi, they have to choose one of the supported prefixes. We can always add more prefixes if we want.

This way simplifies the spec.

Reply:
I _think_ the more modern way of solving the xssi issue is with CORS ([Access-Control-Allow-Origin header](https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS#The_HTTP_response_headers)). We've mainly added this feature for those that cannot modify their backends.

One (small) reason not to always do this is because `response.text().then(JSON.parse)` occurs on the main thread whereas `response.json()` occurs off of it (perf wins).

Reply:
that makes sense

Reply:
please remove it closer to `checkConsentHref`. Also consider giving it a more meaningful name, such as `xssiJsonPrefix`? We should unify the naming across AMP.

Reply:
Pinging @jridgewell here because he has a much better grasp of the attack vectors and mitigations

Reply:
> please remove it closer to checkConsentHref

Sure!

> We should unify the naming across AMP.

The only two other components I think it will live in are `amp-list` and `form`. For both of those our planned name was just `strip-prefix` whereas this one will need to be different since it is part of the json configuration where the convention is camel casing.  We can still aim for a better name everywhere though, like you suggest. Something like: `xssiStripPrefix` and `xssi-strip-prefix`.

Reply:
the new name looks good. Or for simplicity, just `xssiPrefix` & `xssi-prefix`?

Reply:
Done.

Reply:
btw, @lannka, even though I've already made the change here, I'm opening up a discussion for `strip-prefix` vs. `xssi-prefix` since a couple people in my vicinity have stated they preferred the old name 🤷‍♂